File: testng/src/main/java/com/shoppingapp/ProductService.java
package com.shoppingapp;

import java.util.HashMap;
import java.util.Map;

public class ProductService {
    private Map<String, Product> productCatalog = new HashMap<>();

    public void addProduct(Product product) {
        productCatalog.put(product.getName(), product);
    }

    public Product getProductById(String productId) {
        return productCatalog.get(productId);
    }

    public void updateProduct(Product product) {
        productCatalog.put(product.getName(), product);
    }
}

File: behave/src/main/shoppingapp/product_service.py
class ProductService:
    def __init__(self):
        self.product_catalog = {}

    def add_product(self, product):
        self.product_catalog[product.name] = product

    def get_product(self, name):
        return self.product_catalog.get(name)

    def update_product(self, product):
        self.product_catalog[product.name] = product
